csrc/preseg_passes/move_gather.h

```diff
@@ -0,0 +1,24 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <preseg_passes/optimization_pass.h>
+
+namespace nvfuser::preseg_passes {
+
+// A pre-segmenter optimization that moves gather operations ahead of producer
```

Comment:
Please expand the comment. Why should it be done? When is it applicable? Examples would help.